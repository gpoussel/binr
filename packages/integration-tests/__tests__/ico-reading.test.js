"use strict";

const _ = require("lodash");

const fs = require("fs");
const { BinaryReader } = require("@binr/binary-reader");
const { DefinitionReader } = require("@binr/definition-reader");

const pathToBinaryFixtures = `${__dirname}/../__fixtures__/binaries`;
const pathToDefinitionFixtures = `${__dirname}/../__fixtures__/definitions`;

describe("ICO reading", () => {
  const definitionReader = new DefinitionReader();

  const icoDefinitionFile = fs.readFileSync(`${pathToDefinitionFixtures}/ico.binr`);
  const icoDefinition = definitionReader.readInput(icoDefinitionFile.toString());

  const icoFile = `${pathToBinaryFixtures}/bing.ico`;
  const icoBuffer = fs.readFileSync(icoFile);

  test("reads ICO file", () => {
    const binaryReader = new BinaryReader();
    const icoValue = binaryReader.read(icoBuffer, icoDefinition, "IcoFile");
    expect(icoValue).toBeDefined();
    expect(icoValue).toMatchSnapshot();
  });
});
