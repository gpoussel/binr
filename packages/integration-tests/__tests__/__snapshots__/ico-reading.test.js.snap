// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ICO reading reads ICO file 1`] = `
Object {
  "iconDir": Object {
    "count": 2,
    "entries": Array [
      Object {
        "bitCount": 32,
        "bytesInResource": 1128,
        "colorCount": 0,
        "height": 16,
        "imageOffset": 38,
        "planes": 1,
        "reserved": 0,
        "width": 16,
      },
      Object {
        "bitCount": 32,
        "bytesInResource": 4264,
        "colorCount": 0,
        "height": 32,
        "imageOffset": 1166,
        "planes": 1,
        "reserved": 0,
        "width": 32,
      },
    ],
    "reserved": 0,
    "type": 1,
  },
  "images": Array [
    Object {
      "data": Object {},
      "header": Object {
        "bitCount": 32,
        "clrImportant": 0,
        "clrUsed": 0,
        "compression": 0,
        "height": 32,
        "planes": 1,
        "size": 40,
        "sizeImage": 0,
        "width": 16,
        "xPelsPerMeter": 0,
        "yPelsPerMeter": 0,
      },
    },
  ],
}
`;
